src/arrayMethodSort.js

```diff
@@ -1,11 +1,38 @@
+/* eslint-disable no-param-reassign */
 'use strict';
 
 /**
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    // eslint-disable-next-line no-param-reassign
+    compareFunction =
+      compareFunction ||
+      function (a, b) {
+        a = String(a);
+        b = String(b);
+
+        if (a < b) {
+          return -1;
+        }
+
+        if (a > b) {
+          return 1;
+        }
+
+        return 0;
+      };
```

Comment:
instead of this logic, you can just add the default value for the function argument